src/org/labkey/test/params/study/DatasetDefinition.java

```diff
@@ -0,0 +1,131 @@
+package org.labkey.test.params.study;
+
+import org.jetbrains.annotations.NotNull;
+import org.labkey.remoteapi.domain.Domain;
+import org.labkey.remoteapi.domain.PropertyDescriptor;
+import org.labkey.test.params.property.DomainProps;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DatasetDefinition extends DomainProps
+{
+    private String _name;
+    private String _description;
+    private List<PropertyDescriptor> _fields = new ArrayList<>();
+    private String _kindName;
+    private Boolean _demographics;
+    private Boolean _managedKeyField;
+    private String _keyPropertyName;
+    private Boolean _timeKeyField;
+
+    public static final String VISIT_BASED_STUDY = "StudyDatasetVisit";
+    public static final String DATE_BASED_STUDY = "StudyDatasetDate";
+
+    public static DatasetDefinition create(String name)
+    {
+        return new DatasetDefinition(name);
+    }
```

Comment:
This static `create` method is a little confusing because `DomainProps` has a non-static `create` method that actually creates the domain.

Reply:
that's true, I was thinking of maybe `init`, not sure if there is a better verb.